diesel/src/query_builder/select_by_statement/dsl_impls.rs

```diff
@@ -0,0 +1,365 @@
+//             self.where_clause.into(),
+//             self.order.into(),
+//             Box::new(self.limit),
+//             Box::new(self.offset),
+//             Box::new(self.group_by),
+//         )
+//     }
+// }
+
+// impl<'a, F, D, W, O, L, Of, G, DB> BoxedDsl<'a, DB>
+//     for SelectStatement<F, DefaultSelectClause, D, W, O, L, Of, G>
+// where
+//     Self: AsQuery,
+//     DB: Backend,
+//     F: QuerySource,
+//     F::DefaultSelection: QueryFragment<DB> + Send + 'a,
+//     D: QueryFragment<DB> + Send + 'a,
+//     W: Into<BoxedWhereClause<'a, DB>>,
+//     O: Into<Option<Box<dyn QueryFragment<DB> + Send + 'a>>>,
+//     L: QueryFragment<DB> + Send + 'a,
+//     Of: QueryFragment<DB> + Send + 'a,
+//     G: QueryFragment<DB> + Send + 'a,
+// {
+//     type Output = BoxedSelectStatement<'a, <F::DefaultSelection as Expression>::SqlType, F, DB>;
+//     fn internal_into_boxed(self) -> Self::Output {
+//         BoxedSelectStatement::new(
+//             Box::new(self.from.default_selection()),
+//             self.from,
+//             Box::new(self.distinct),
+//             self.where_clause.into(),
+//             self.order.into(),
+//             Box::new(self.limit),
+//             Box::new(self.offset),
+//             Box::new(self.group_by),
+//         )
+//     }
+// }
+
+impl<S, STMT> HasTable for SelectByStatement<S, STMT>
+where
+    STMT: HasTable,
+{
+    type Table = STMT::Table;
+
+    fn table() -> Self::Table {
+        STMT::table()
+    }
+}
+
+// no IntoUpdateTarget: if you'd like to `.into_update_target`, you should not first `.select_by`
+// impl<F, W> IntoUpdateTarget for SelectStatement<F, DefaultSelectClause, NoDistinctClause, W>
+// where
+//     SelectStatement<F, DefaultSelectClause, NoDistinctClause, W>: HasTable,
+//     W: ValidWhereClause<F>,
+// {
+//     type WhereClause = W;
+
+//     fn into_update_target(self) -> UpdateTarget<Self::Table, Self::WhereClause> {
+//         UpdateTarget {
+//             table: Self::table(),
+//             where_clause: self.where_clause,
+//         }
+//     }
+// }
+
+// FIXME: Should we disable joining when `.group_by` has been called? Are there
+// any other query methods where a join no longer has the same semantics as
+// joining on just the table?
+impl<S, STMT, Rhs> JoinTo<Rhs> for SelectByStatement<S, STMT>
+where
+    STMT: JoinTo<Rhs>,
+{
+    type FromClause = STMT::FromClause;
+    type OnClause = STMT::OnClause;
+
+    fn join_target(rhs: Rhs) -> (Self::FromClause, Self::OnClause) {
+        STMT::join_target(rhs)
+    }
+}
+
+impl<S, STMT> QueryDsl for SelectByStatement<S, STMT> {}
+
+impl<S, STMT, Conn> RunQueryDsl<Conn> for SelectByStatement<S, STMT> {}
+
+impl<S, STMT, Tab> Insertable<Tab> for SelectByStatement<S, STMT>
+where
+    Tab: Table,
+    Self: Query,
+{
+    type Values = InsertFromSelect<Self, Tab::AllColumns>;
+
+    fn values(self) -> Self::Values {
+        InsertFromSelect::new(self)
```

Comment:
I think its sufficient here to just pass `self.inner` to the `InsertFromSelect` clause, as it's the same query and noone can modify the select clause there.